tools/state-viewer/src/dump_state_parts.rs

```diff
@@ -1,15 +1,96 @@
+use crate::epoch_info::iterate_and_filter;
+use clap::Subcommand;
 use near_chain::{ChainStore, ChainStoreAccess, RuntimeAdapter};
+use near_epoch_manager::EpochManager;
+use near_primitives::epoch_manager::epoch_info::EpochInfo;
 use near_primitives::state_part::PartId;
 use near_primitives::syncing::get_num_state_parts;
+use near_primitives::types::EpochId;
 use near_primitives_core::hash::CryptoHash;
-use near_primitives_core::types::ShardId;
+use near_primitives_core::types::{BlockHeight, EpochHeight, ShardId};
 use near_store::Store;
 use nearcore::{NearConfig, NightshadeRuntime};
 use std::path::Path;
+use std::str::FromStr;
 use std::sync::Arc;
 
+#[derive(Subcommand, Debug, Clone)]
+pub(crate) enum EpochSelection {
+    /// Current epoch.
+    Current,
+    /// Fetch the given epoch.
+    EpochId { epoch_id: String },
+    /// Fetch epochs at the given height.
+    EpochHeight { epoch_height: EpochHeight },
+    /// Fetch an epoch containing the given block hash.
+    BlockHash { block_hash: String },
+    /// Fetch an epoch containing the given block height.
+    BlockHeight { block_height: BlockHeight },
+}
+
+fn get_epoch_id(
```

Comment:
nit:

I'd actually make this function as 'impl' function of EpochSelector:

impl EpochSelector {
pub to_epoch_id(store, chain_store, epoch_maanger) -> EpochId